From 24e77f263b9a8353a4705593ebe1c3211835eef4 Mon Sep 17 00:00:00 2001
From: Satyam Kumar Singh <quic_satysing@quicinc.com>
Date: Thu, 4 Sep 2025 04:39:49 +0000
Subject: [PATCH] [PATCH] View: Disable VRR feature to meet power expectations.

VRR wakes up the idle device after 750ms timeout. This bounds Display
and GPU to wake up as well, leading to higher power consumption during
idling. Disable VRR to avoid this extra wake up call as it can cause
power regression.
Google bug: 409971466

Change-Id: I699434940dfaf5eb5e5d3577a7eea335a39d0c44
---
 core/java/android/view/ViewRootImpl.java | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/core/java/android/view/ViewRootImpl.java b/core/java/android/view/ViewRootImpl.java
index c1b92ee3f74e..66b8e7d5d6f9 100644
--- a/core/java/android/view/ViewRootImpl.java
+++ b/core/java/android/view/ViewRootImpl.java
@@ -1223,7 +1223,12 @@ public final class ViewRootImpl implements ViewParent,
             Flags.enableInvalidateCheckThread();
     private static boolean sSurfaceFlingerBugfixFlagValue =
             com.android.graphics.surfaceflinger.flags.Flags.vrrBugfix24q4();
-    private static final boolean sEnableVrr = ViewProperties.vrr_enabled().orElse(true);
+    // Disable VRR feature to meet power expectations.
+    // VRR wakes up the idle device after 750ms timeout. This bounds Display and GPU to
+    // wake up as well, leading to higher power consumption during idling.
+    // Disable VRR to avoid this extra wakeup call as it can cause power regression.
+    // Google bug: https://partnerissuetracker.corp.google.com/u/0/issues/409971466
+    private static final boolean sEnableVrr = false;
 
     static {
         sToolkitSetFrameRateReadOnlyFlagValue = toolkitSetFrameRateReadOnly();
-- 
2.34.1

